File: frontend/src/features/commerce/orderStatus.js
const NEXT_STATUSES = {
  pending: ["processing", "cancelled"],
  processing: ["shipped", "cancelled"],
  shipped: ["completed"],
  completed: [],
  cancelled: [],
};

const STATUS_LABEL_KEYS = {
  pending: "statusPending",
  processing: "statusProcessing",
  shipped: "statusShipped",
  completed: "statusCompleted",
  cancelled: "statusCancelled",
};

export function nextOrderStatuses(status) {
  return NEXT_STATUSES[status] || [];
}

export function orderStatusLabelKey(status) {
  return STATUS_LABEL_KEYS[status] || "statusUnknown";
}

File: backend/app/order_status.py
ORDER_STATUSES = frozenset({"pending", "processing", "shipped", "completed", "cancelled"})

ORDER_STATUS_TRANSITIONS = {
    "pending": {"processing", "cancelled"},
    "processing": {"shipped", "cancelled"},
    "shipped": {"completed"},
    "completed": set(),
    "cancelled": set(),
}


def can_transition(current: str, requested: str) -> bool:
    return current == requested or requested in ORDER_STATUS_TRANSITIONS.get(current, set())
